app/bundles/CoreBundle/Translation/Translator.php

```diff
@@ -18,58 +72,39 @@ class Translator extends BaseTranslator
      *
      * @return bool true if the message has a translation, false otherwise
      */
-    public function hasId($id, $domain = null, $locale = null)
+    public function hasId(string $id, ?string $domain = null, ?string $locale = null): bool
     {
-        if (null === $locale) {
-            $locale = $this->getLocale();
-        } else {
-            $this->assertValidLocale($locale);
-        }
-
         if (null === $domain) {
             $domain = 'messages';
         }
 
-        if (!isset($this->catalogues[$locale])) {
-            $this->loadCatalogue($locale);
-        }
-
-        return $this->getCatalogue($locale)->has((string) $id, $domain);
+        return $this->getCatalogue($locale)->has($id, $domain);
     }
 
     /**
-     * {@inheritdoc}
+     * Checks for $preferred string existence and returns translation if it does.  Otherwise, returns translation for
+     * $alternative.
+     *
+     * @param array<mixed> $parameters
      */
-    protected function loadCatalogue($locale)
+    public function transConditional(string $preferred, string $alternative, array $parameters = [], ?string $domain = null, ?string $locale = null): string
     {
-        if ('en_US' != $locale) {
-            // Always force en_US so that it's available for fallback
-            $this->addResource('mautic', null, 'en_US', 'messages');
+        if ($this->hasId($preferred, $domain, $locale)) {
+            return $this->trans($preferred, $parameters, $domain, $locale);
         }
 
-        $this->addResource('mautic', null, $locale, 'messages');
-
-        parent::loadCatalogue($locale);
+        return $this->trans($alternative, $parameters, $domain, $locale);
     }
 
     /**
-     * Checks for $preferred string existence and returns translation if it does.  Otherwise, returns translation for
-     * $alternative.
+     * Passes through all unknown calls onto the translator object.
      *
-     * @param      $preferred
-     * @param      $alternative
-     * @param      $parameters
-     * @param null $domain
-     * @param null $locale
+     * @param mixed $args
      *
-     * @return string
+     * @return mixed
      */
-    public function transConditional($preferred, $alternative, $parameters = [], $domain = null, $locale = null)
+    public function __call(string $method, $args)
```

Comment:
Why do we need this magic?

Reply:
Because this class decorates given `translator` service and have no clue what public methods are available in the class.

Reply:
What other methods are used on the Translator? It implements all methods from the interfaces and some more. The dependency translator expects only to have the `trans()` method. I wouldn't provide any more wiggle room than it already has. Magic methods make troubles for static analysis.

Reply:
See how they did it https://github.com/symfony/translation/blob/6.1/DataCollectorTranslator.php#L115
All public methods of the decorating service must be implemented, otherwise the code will give MethodNotFound in unexpected places. But as we do not know which methods are in the `translator` service, then we are forced to use `__call` here

Reply:
Thanks for the link. I'm still not convinced though. Is something failing without it? How about we try it without it and if something comes up we'll deal with it then? I'd like to avoid adding magic methods to the code just because they might be useful one day or because someone else do it.

Reply:
@biozshock would you mind if I remove the magic method and resolve the conflict so we can get this tested and merged?

Reply:
@escopecz removed in [03035f8](https://github.com/mautic/mautic/pull/11447/commits/03035f816be82162f29d6c873c56440dc6a1491a)